common/lib/xmodule/xmodule/css/capa/display.scss

```diff
@@ -906,47 +905,79 @@ div.problem {
 // +Problem - Actions
 // ====================
 div.problem .action {
-  margin-top: $baseline;
+  @include margin($baseline 0 $baseline 0);
+  min-height: $baseline;
 
-  .save, .check, .show, .reset, .hint-button {
-    @include margin-right($baseline/2);
-    margin-bottom: ($baseline/2);
-    height: ($baseline*2);
-    vertical-align: middle;
-    text-transform: uppercase;
-    font-weight: 600;
-    @media print {
-      display: none;
+  // override default button hover
+  button {
+    &:hover {
+      background-image: none;
+      box-shadow: none;
+    }
+    &:focus {
+      box-shadow: none;
+    }
+    &.btn-default {
+      background-color: transparent;
+    }
+    &.btn-brand {
+      &:hover {
+        background-color: #0ea6ec;
+      }
+      &.check {
+        &.is-disabled, &.disabled {
+          background: transparent;
+          border-color: #d2d0d0;
+          color: #6b6969;
+        }
+      }
     }
   }
 
-  .save {
-    @extend .blue-button !optional;
+  @media (min-width: 768px) {
+
+    .problem-action-buttons-wrapper {
+      position: absolute;
+      @include margin-right(($baseline*.75));
+      right: 0;
+    }
+
   }
 
-  .show {
+  .problem-action-button-wrapper {
+    @include padding(($baseline*.75) 0 ($baseline/2) ($baseline/4));
+    border-right: 1px solid #ddd;
+    &:last-child {
+      border: none;
+    }
```

Comment:
This is a good example of how to order the different kinds of things I was talking about above!